scripts/sync-gsuite.js

```diff
@@ -0,0 +1,233 @@
+
+// PART 1: get all gsuite under /individuals, and for each one make a query to core
+
+const gsuite_obj = [
+  // {
+  //   "primaryEmail": name.surname@aegee.eu, // from gsuite
+  //   "recoveryEmail": name.surname@gmail.com, // from gsuite
+  //   "g_surname": item.name.familyName, // surname coming from gsuite
+  //   "m_possible_users": [], // heuristic of people with the same surname in myaegee
+  //   "allocated": false, // whether we have found this person on myaegee
+  // } ;
+];
+
+//GAM tells me there are 800-900 users, so we need two invocations.
+
+const superagent = require('superagent');
+
+(async () => {
+
+  try{
+
+    const part1 = await superagent.get('gsuite-wrapper:8084/accounts?max=500')
+
+    const part2 = await superagent.get('gsuite-wrapper:8084/accounts?max=500&pageToken='+part1.body.data.nextPageToken)
+
+    let unrecovery=0
+
+    part1.body.data.users.concat(part2.body.data.users).forEach( (item) => {
+      gsuite_obj.push(
+      {
+        "primaryEmail": item.primaryEmail,
+        "recoveryEmail": item.recoveryEmail || item.emails[0].address,
+        "g_surname": item.name.familyName,
+        "m_possible_users": [],
+        "allocated": false
+      }) ;
+
+      unrecovery += item.recoveryEmail ? 0 : 1 ;
+
+    });
+
+    //double check
+    let unsecondary=0
+    gsuite_obj.filter(obj => obj.recoveryEmail==='').forEach( obj => { unsecondary +=1; console.log(obj.primaryEmail+" has no secondary email!") } );
+    console.log("There are "+unrecovery+" users without a proper recovery email, only secondary; and without secondary are "+unsecondary)
+  }catch (e){
+    console.log(e)
+  }
+})();
+
+async function findMyAEGEEuser(obj) {
+
+  const users = await User.findAll({
+    where: { email: obj.recoveryEmail.toLowerCase() }
+  });
+
+//PART 2: if there is a match, add their gsuite address in the core's field and mark it done
+  if(users.length === 1 ){
+    try{
+      const wow = await User.update({
+        gsuite_id: obj.primaryEmail
+      },
+      {
+        where: { email: obj.recoveryEmail.toLowerCase() },
+        returning: true,
+        plain: true
+      });
+
+      obj.allocated=true
+    }catch (e){
+      console.log(e)
+    }
+  }
+}
+
+gsuite_obj.forEach( findMyAEGEEuser )
+
+//PART 3: now look at groups
```

Comment:
Why would we look at groups again? Adding people to the correct groups is something we'll do later when we sync MyAEGEE circles with groups

Reply:
yep in fact it's a stub, to be implemented